sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/main/java/org/apache/shardingsphere/orchestration/center/instance/wrapper/ApolloOpenApiWrapper.java

```diff
@@ -0,0 +1,108 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.orchestration.center.instance.wrapper;
+
+import com.ctrip.framework.apollo.core.ConfigConsts;
+import com.ctrip.framework.apollo.openapi.client.ApolloOpenApiClient;
+import com.ctrip.framework.apollo.openapi.client.constant.ApolloOpenApiConstants;
+import com.ctrip.framework.apollo.openapi.dto.NamespaceReleaseDTO;
+import com.ctrip.framework.apollo.openapi.dto.OpenItemDTO;
+import com.google.common.primitives.Ints;
+import org.apache.shardingsphere.orchestration.center.configuration.InstanceConfiguration;
+
+import java.util.Objects;
+import java.util.Properties;
+
+/**
+ * Apollo open api client wrapper.
+ *
+ * @author dongzonglei
+ */
+public final class ApolloOpenApiWrapper {
+    
+    private ApolloOpenApiClient client;
+    
+    private String namespace;
+    
+    private String appId;
+    
+    private String env;
+    
+    private String clusterName;
+    
+    private String administrator;
+    
+    public ApolloOpenApiWrapper(final InstanceConfiguration config, final Properties properties) {
+        namespace = config.getNamespace();
+        appId = properties.getProperty("appId", "APOLLO_SHARDING_SPHERE");
+        env = properties.getProperty("env", "DEV");
+        clusterName = properties.getProperty("clusterName", ConfigConsts.CLUSTER_NAME_DEFAULT);
+        administrator = properties.getProperty("administrator");
+        String apolloToken = properties.getProperty("token");
+        String portalUrl = properties.getProperty("portalUrl");
+        Integer connectTimeout = Ints.tryParse(Objects.toString(properties.get("connectTimeout")));
+        Integer readTimeout = Ints.tryParse(Objects.toString(properties.get("readTimeout")));
+        client = ApolloOpenApiClient.newBuilder().withPortalUrl(portalUrl)
+                .withConnectTimeout(connectTimeout == null ? ApolloOpenApiConstants.DEFAULT_CONNECT_TIMEOUT : connectTimeout)
+                .withReadTimeout(readTimeout == null ? ApolloOpenApiConstants.DEFAULT_READ_TIMEOUT : readTimeout)
+                .withToken(apolloToken).build();
+    }
+    
+    /**
+     * Get config value by key.
+     * 
+     * @param key key
+     * @return value
+     */
+    public String getValue(final String key) {
+        OpenItemDTO itemDTO = client.getItem(appId, env, clusterName, namespace, key);
+        if (itemDTO == null) {
+            return null;
+        }
+        return itemDTO.getValue();
+    }
+    
+    /**
+     * Persist config.
+     * 
+     * @param key key
+     * @param value value
+     */
+    public void persist(final String key, final String value) {
```

Comment:
Whether we should add TCC code for transaction.

Reply:
fix code, use try catch.